.circleci/config.yml

```diff
@@ -0,0 +1,130 @@
+# THIS IS A CUSTOM CIRCLE CONFIG, IT WAS NOT GENERATED
+version: 2
+
+references:
+
+  container_config_node8: &container_config_node8
+    working_directory: ~/project/build
+    docker:
+      - image: circleci/node:8
+
+  workspace_root: &workspace_root
+    ~/project
+
+  attach_workspace: &attach_workspace
+    attach_workspace:
+      at: *workspace_root
+
+  npm_cache_keys: &npm_cache_keys
+    keys:
+      - v1-dependency-npm-{{ checksum "package.json" }}
+      - v1-dependency-npm-
+
+  cache_npm_cache: &cache_npm_cache
+    save_cache:
+      key: v1-dependency-npm-{{ checksum "package.json" }}
+      paths:
+        - ./node_modules/
+        - ./tools/x-docs/node_modules/
+        - ./tools/x-workbench/node_modules/
```

Comment:
I'd suggest you have a separate cache for each `node_modules` directory, otherwise things could get messy. Example of multiple caches: https://circleci.com/docs/2.0/caching/#full-example-of-saving-and-restoring-cache

Reply:
Done!